src/cdsetool/download.py

```diff
@@ -158,3 +164,31 @@ def _get_credentials(options: Dict) -> Credentials:
 
 def _get_temp_dir(options: Dict) -> Union[str, None]:
     return options.get("tmpdir") or None
+
+
+def checksum_for_file(
+    file_path: str, file_name: str, proxy: Union[Dict[str, str], None] = None
+) -> bool:
+    """Evaluate the checksum of a file. If the file was downloaded correctly"""
+    odata_for_checksum = get_odata_by_name(file_name, proxy)
+    array_values = odata_for_checksum.get("value", [])
+    if len(array_values) == 0:
+        # no data available for checksum
+        return False
+    for value in array_values:
+        if value["name"] != file_name:
+            continue
+        checksums: list = value["Checksum"]
+        for checksum in checksums:
+            if checksum.get("Algorithm", "") == "MD5":
```

Comment:
Does something like the following work instead of the current body of the loop?
```python
            match checksum.get("Algorithm"):
                case "MD5":
                    calculated_sum = hashlib.md5(file_path.read()).hexdigest()
                case "SHA3-256":
                    calculated_sum = hashlib.sha3_256(file_path.read()).hexdigest()
                case algorithm:  # pragma: no cover
                    raise ValueError(f"Unexpected checksum algorithm '{algorithm}'")
            expected_sum = checksum.get("Value", "")
            if calculated_sum != expected_sum:
                log.warning(f"Expected checksum {expected_sum} but got {calculated_sum} for {file_path}")
               return False
````

Reply:
Actually I discovered that the match\case statement is was introduced in Python3.10 so the tests on python 3.9 fails. So let me know if we use the if elif clause or mantain the match\case.

Reply:
Oh, I have no idea what Python version that introduced something, sorry about that! Keep the if-then-else.

The code in your reply to my comment loops through the checksums until it finds an acceptable checksum. I don't trust the metadata, so I'm wondering if the logic should instead be to try checksums in order of algorithmic preference, so we define something like (pseudo-code):
```python
if not checksums:
  return True
for name, fun  in [("BLAKE", hashlib.blake), ("SHA3-256", hashlib.sha3_256), ("MD5", hashlib_md5)]:
  expected_sum = get_expected_checksum(algorithm, checksums)
  if expected_sum:
    with open(file_path, "rb") as file:
      calculated_sum = fun(file.read()).hexdigest()
      return calculated_sum == expected_sum
return False
```
so we get a failure if there is a checksum for our algorithm of choice that doesn't match the file we have downloaded, and we also get a failure if there are checksums for the archive but only with algorithms we don't support.